webapp/components/survey/CategoryDetails/CategoryDetails.js

```diff
@@ -15,35 +15,32 @@ import { useAuthCanEditSurvey } from '@webapp/store/user'
 import { FormItem, Input } from '@webapp/components/form/input'
 import UploadButton from '@webapp/components/form/uploadButton'
 
-import CategoryImportSummary from './components/categoryImportSummary'
-import LevelEdit from './components/levelEdit'
+import ImportSummary from './ImportSummary'
+import LevelEdit from './LevelEdit'
 
-import * as CategoryState from './categoryState'
-import * as Actions from './actions'
+import * as Actions from '../../../loggedin/surveyViews/category/actions'
+import { State, useLocalState } from './store'
+
+const CategoryDetails = (props) => {
+  const { showClose, onCategoryCreated, categoryUuid: categoryUuidProp } = props
 
-const CategoryView = (props) => {
-  const { showClose } = props
   const { categoryUuid } = useParams()
   const i18n = useI18n()
   const dispatch = useDispatch()
   const history = useHistory()
 
-  const category = useSelector(CategoryState.getCategoryForEdit)
-  const importSummary = useSelector(CategoryState.getImportSummary)
   const readOnly = !useAuthCanEditSurvey()
 
+  const { state } = useLocalState({ onCategoryCreated, categoryUuid: categoryUuidProp })
```

Comment:
a doubt: shouldn't categoryUuid be the one from props or the one from url params?

Reply:
in this case it's the one coming from the props. The one from the params won't be handled by the CategoryDetails component, but in useLocalState and the property inCategoriesPath will be in the CategoryDetails local state.